vita3k/app/src/app_init.cpp

```diff
@@ -101,11 +102,92 @@ void update_viewport(EmuEnvState &state) {
     }
 }
 
+void init_paths(Root &root_paths) {
+    auto base_path = SDL_GetBasePath();
+    auto pref_path = SDL_GetPrefPath(org_name, app_name);
+    root_paths.set_base_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_pref_path(string_utils::utf_to_wide(pref_path));
+    root_paths.set_log_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_config_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_shared_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_cache_path(fs::path(string_utils::utf_to_wide(base_path)) / "cache");
+    SDL_free(base_path);
+    SDL_free(pref_path);
+
+#if defined(__linux__) && !defined(__ANDROID__) && !defined(__APPLE__)
+    // XDG Data Dirs.
+    auto env_home_dir = getenv("HOME");
+    auto env_data = getenv("XDG_DATA_HOME");
+    auto env_home_data = env_data;
+    if (env_data != NULL)
+        root_paths.set_pref_path(string_utils::utf_to_wide(env_data) / fs::path(app_name) / fs::path(app_name));
+
+    root_paths.set_log_path(root_paths.get_pref_path());
+
+    env_data = getenv("XDG_CONFIG_HOME");
+    if (env_data != NULL)
+        root_paths.set_config_path(string_utils::utf_to_wide(env_data) / fs::path(app_name));
+    else if (env_home_dir != NULL)
+        root_paths.set_config_path(fs::path(string_utils::utf_to_wide(env_home_dir)) / ".config" / fs::path(app_name));
+
+    env_data = getenv("XDG_CACHE_HOME");
+    if (env_data != NULL)
+        root_paths.set_cache_path(string_utils::utf_to_wide(env_data) / fs::path(app_name));
+    else if (env_home_dir != NULL)
+        root_paths.set_cache_path(fs::path(string_utils::utf_to_wide(env_home_dir)) / ".cache" / fs::path(app_name));
+
+    if (env_home_dir != NULL) {
```

Comment:
continuation from the comment about $HOME, if you remove this check then shared path will be `~/.local/share/Vita3K` by default, so the code in line 155 if ran would do the same thing, setting it to the same path again

Reply:
also you can replace all uses of env_home_dir with `~` as its an alias for $HOME

Reply:
Fixed.

Reply:
> also you can replace all uses of env_home_dir with `~` as its an alias for $HOME

Changing the resolved HOME env var to "~/" results in a boost::filesystem exception (boost::filesystem::copy_file: Invalid cross-device link) on pkg install: 
![Screenshot_2023-10-14_07-25-46](https://github.com/Vita3K/Vita3K/assets/137227028/e6a61ebf-9d21-4c2b-a7b6-39730c2ee2dd)

So I'll probably need to revert this part.